Makefile

```diff
@@ -12,15 +12,23 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+DOCKER ?= docker
+GOLANG_VERSION ?= 1.15.6
+BUILDIMAGE ?= nvidia/go-nvml-devel
+
 GEN_DIR = $(PWD)/gen
 PKG_DIR = $(PWD)/pkg
 GEN_BINDINGS_DIR = $(GEN_DIR)/nvml
 PKG_BINDINGS_DIR = $(PKG_DIR)/nvml
 
 SOURCES = $(shell find $(GEN_BINDINGS_DIR) -type f)
 
-.PHONY: all test clean
-.PHONY: bindings test-bindings clean-bindings
+.PHONY: $(TARGETS)
+TARGETS := $(NATIVE_TARGETS) $(DOCKER_TARGETS)
+NATIVE_TARGETS := all test clean bindings test-bindings clean-bindings
+DOCKER_TARGETS := $(patsubst %, docker-%, $(NATIVE_TARGETS))
+
+$(DOCKER_TARGETS): %: docker-%
```

Comment:
It seems unnecessary to have this.

I think the target below for `docker-%` can just be changed to `$(DOCKER_TARGETS)`.

Reply:
Let me see. It's been a while since I've had to write real make files, so I could definitely improve things.